making parallel execution of the test in test_uzfs.sh

mynktl opened the pull request:
#### Changes in test_uzfs.sh
          - removing jobs related to test_type in Travis file.
          - setting build branch to ZFS-0.7-release for Travis build
          - display log if any test case fails
          - removed init_test/close_test function
          - Added restart of ZREPL before fio_test in test_uzfs.sh to avoid bind error in FIO
#### removing pool/dataset creation in rebuild test
#### fixing multiple free of snap_name in case of error in uzfs_get_io_diff.
#### fixing SIGSEGV error in uzfs_spa_fini.
          - If POOL is busy then we should not destroy spa->spa_us.
#### fixing race condition in uzfs_spa_fini.
          - If there are two threads calling spa_close at a time, then there are chances
            of race condition in uzfs_spa_fini, which causes SIGSEGV in uzfs_spa_fini.
#### fix in spa_fini->spa_evict_all related to uzfs_spa_fini
          - Changing _UZFS macro to _KERNEL macro for uzfs_spa_fini as 
            uzfs_spa_init is protected through _KERNEL macro only
          - txg_update_thread cleanup is being handled in uzfs_spa_fini which
            was being called after spa_deactivate in spa_evict_all.
            In txg_update_thread, dsl_sync_task opens pool which performs
            spa_open->uzfs_spa_init if spa state is POOL_STATE_UNINITIALIZED.
            Since, pool state is already set to POOL_STATE_UNINITIALIZED by spa_deactivate,
            txg_update_thread will execute uzfs_spa_init and that cause
            uzfs_spa_fini waits for infinite time for txg_update_thread cleanup.
### Changes in uzfs_zvol_zap.c for uzfs zap operation test
          - changed sleep time according to txg_update_interval_time to check if 
            LAST_ITER_TXG is being updated or not. Also added txg_wait_synced
            to force txg to change.

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

Commit:
- making parallel execution of the test in test_uzfs.sh
- removing jobs related to test_type in Travis file.
- setting build branch to zfs-0.7-release for Travis build
- removing pool/dataset creation in rebuild test

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

jkryl commented on tests/cbtest/script/test_uzfs.sh:
```diff
@@ -60,16 +57,74 @@ log_note()
+
+	if [ $status -eq 0 ]; then
+		log_fail $@
+	fi
```
this function is same as the previous one except last few lines. It would have been good if we put common part to a new function and share the code between them. Also if the return status of the command is ok, we could hide the output to make the travis log file better readable.

Commit:
- fixing multiple free of snap_name in case of error in uzfs_get_io_diff.
- fixing SIGSEGV error in uzfs_spa_fini.
	- If pool is busy then we should not destroy spa->spa_us.
- fixing race condition in uzfs_spa_fini.
	- If there are two threads calling spa_close at a time, then there
	  are chances of race condition in uzfs_spa_fini.
- Changes in test_uzfs.sh
	- display log if any test case fails
	- removed init_test/close_test function

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

Commit:
- disabling zrepl for gtest related tests

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

Commit:
- fix in spa_fini->spa_evict_all related to uzfs_spa_fini
	txg_update_thread cleanup is being handled in uzfs_spa_fini which
	was being called after spa_deactivate in spa_evict_all.
	In txg_update_thread, dsl_sync_task opens pool which performs
	spa_open->uzfs_spa_init if spa state is POOL_STATE_UNINITIALIZED.
	Since, pool state is already set to POOL_STATE_UNINITIALIZED by spa_deactivate,
	txg_update_thread will execute uzfs_spa_init and that cause
	uzfs_spa_fini waits for infinite time for txg_update_thread cleanup.

- Added restart of zrepl before fio_test in test_uzfs.sh to avoid bind error in fio
- releasing spa_namespace_lock in uzfs_spa_fini before sleep to avoid deadlock with
  txg_update_thread

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

Commit:
Merge remote-tracking branch 'upstream/zfs-0.7-release' into cstor139

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

Commit:
- removing static path for FIO_SRC in test_uzfs.sh
- enabling parallel build in travis

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

vishnuitta commented on cmd/uzfs_test/uzfs_zvol_zap.c:
```diff
@@ -170,7 +178,7 @@ uzfs_zvol_zap_operation(void *arg)
 
 	zfs_txg_timeout = 1;
-	ts.tv_sec = 3;
+	ts.tv_sec = 5;
```
this kind of timebased ones can again break in future. should we have some other way of verifying them? and also, this comment is not needed to be taken up in this PR.

mynktl replied:
ok.

vishnuitta commented on lib/libzpool/uzfs_mgmt.c:
```diff
@@ -496,14 +496,15 @@ uzfs_spa_init(spa_t *spa)
 		    TS_RUN, 0, PTHREAD_CREATE_DETACHED);
 		mutex_exit(&us->mtx);
+	} else {
+		spa->spa_us = us;
```
'us' is not allocated here. if intent is to initialize to NULL, initialize it before 'if'

mynktl replied:
ok. spa is allocated through kmem_zalloc, so we can avoid initializing it to NULL here.

Commit:
- addressing review comments
- changes in test_uzfs.sh
       - exporting pool in test_uzfs.sh before executing UZFS_TEST binary.
	 As, pool should be used by one process only.
       - added start_time and end_time in test_uzfs.sh logs.
       - fix in run_uzfs_test function as some tests were not being executed
	 due to improper if..else statement
- fix in spa_evict_all regarding uzfs_spa_fini. Since uzfs_spa_init was
  protected through _KERNEL macro, uzfs spa_fini also should be protected
  through _KERNEL macro only.
- enabling ztest in travis kernel build.

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

Commit:
- changes in lib/fio/replica.c according to fio source code

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

vishnuitta committed:
Merge branch 'zfs-0.7-release' into cstor139
